types/proposal_test.go

```diff
@@ -235,23 +238,50 @@ func TestIsTimely(t *testing.T) {
 			proposalHeight: 2,
 			proposalTime:   genesisTime.Add(4 * time.Nanosecond),
 			recvTime:       genesisTime,
+			round:          0,
 			precision:      time.Nanosecond * 2,
 			msgDelay:       time.Nanosecond,
 			expectTimely:   false,
 		},
+		{
+			/*
+				With adaptive synchrony params after 4 rounds, MSGDELAY(4) = 1.4641 * 2000ns
+				Recv time must not exceed proposalTime + 2928ns (msgDelay) + 50ns (prcision) = 2978ns
+			*/
+			name:           "proposal time too large for round",
+			proposalHeight: 2,
+			proposalTime:   genesisTime,
+			recvTime:       genesisTime.Add(2980 * time.Nanosecond),
+			round:          4,
+			precision:      time.Nanosecond * 50,
+			msgDelay:       time.Nanosecond * 2000,
+			expectTimely:   false,
+		},
+		{
+			name:           "proposal timely for round",
+			proposalHeight: 2,
+			proposalTime:   genesisTime,
+			recvTime:       genesisTime.Add(2970 * time.Nanosecond),
+			round:          4,
+			precision:      time.Nanosecond * 50,
+			msgDelay:       time.Nanosecond * 2000,
+			expectTimely:   true,
```

Comment:
Are these tests based in a find-grained timing behavior? I'm concerned they'll become flaky when run in heavy-loaded systems (e.g., on GitHub runners)

Reply:
There is no time aspect here. It is only comparing two predefined time instants. It could be any time instant.

Reply:
If that's the case, please disregard my comment